### ISSUE TITLE
Question: Difference in initial loading of tiles images between versions of OSD

### ISSUE THREAD

**Issue by BeebBenjamin** [openseadragon/openseadragon#2456]
I've overriden Mirador with version 4.1.0 of OSD.  One of my team noticed that there is a difference in the intial load of images.

Version 2.4.2 leaves a white space whilst loading initial tiles and loads the image in full.

Version 4.1.0 appears to load tiles in and draws them immediately, so people can see the tiles loading in.  Is this intentional?

![Screenshot from 2024-01-22 14-13-53](https://github.com/openseadragon/openseadragon/assets/4963887/3d486e4c-18cb-4465-b413-4c3331937f6d)

**Comment by iangilman** [comment]
The effect you're referring to sounds like `immediateRender`. That's always been available but always turned off by default. That hasn't changed in the new version of OSD. You can see it's not happening on https://openseadragon.github.io/. Perhaps somehow it got turned on in your project?

**Comment by BeebBenjamin** [comment]
When I set immediateRender to `true` in the osdConfig, the above behaviour stops.

**Comment by iangilman** [comment]
Strange! I would've expected the opposite. I'm not sure what is going on there.

I suppose this resolves your issue, though? If you'd like to dig into it further, it would be helpful if you could share a repro.

**Comment by BeebBenjamin** [comment]
Using Mirador 3.0.0 overriden with OSD version 4.1.0, Mirador has OSD version 2.4.2 code in its minified build.

I've had a look at what happens when setting `immediateRender`.

If false example values

```json
{
"optimalRatio": 6.512676056338028, 
"levelOpacity": 0.628169014084506, 
"levelVisibility": 1.1428571428571428
},
{
"levelOpacity": 0.628169014084506,
"levelVisibility": 1.1428571428571428,
"optimalRatio": 6.512676056338028
},
{
"optimalRatio": 6.512676056338028, 
"levelOpacity": 1, 
"levelVisibility": 1.3333333333333335
},
{
"optimalRatio": 6.512676056338028, 
"levelOpacity": 1, 
"levelVisibility": 2
},
{
"optimalRatio": 6.512676056338028, 
"levelOpacity": 1, 
"levelVisibility": "Infinity"
}
```
If true example values

```json
{
"optimalRatio": 1, 
"levelOpacity": 0.628169014084506, 
"levelVisibility": 5.37878787878788
},
{
"optimalRatio": 1, 
"levelOpacity": 1, 
"levelVisibility": 1.5919282511210762
},
{
"optimalRatio": 1, 
"levelOpacity": 1, 
"levelVisibility": 0.4431960049937578
}
```

**Comment by iangilman** [comment]
Cool. I'm actually not sure what those values should be... That may be normal. Certainly with `false` considering more tiles makes sense. The `Infinity` is interesting... I don't know if that's normal or not.

**Comment by BeebBenjamin** [comment]
Yeah the infinity only appears when it is set to false, never when it is set to true.  I think this piece of code controls blending.  I am perplexed that the fix is something we didn't have to do previously and the default is the same!  I'll take another look at a diff or something comparing versions.

**Comment by iangilman** [comment]
Yeah, it's pretty weird! It's definitely not the case with these versions of OSD running on their own... This "switch" isn't something that happens normally between these versions, so there must be something in how you or Mirador are using it. It would be interesting to see if you could make a simple repro outside of your project, even outside of Mirador.

**Comment by BeebBenjamin** [comment]
The behaviour for this has changed.  Setting immediateRender now does nothing, also setting it to false does nothing.

**Comment by BeebBenjamin** [comment]
I just pulled the latest master to build from and noticed this change.

**Comment by BeebBenjamin** [comment]
I reverted: https://github.com/openseadragon/openseadragon/pull/2537 locally and this restored the behaviour I was seeing, tiles loading before being drawn.  So something about this PR changed tiled loading for me, as I am getting the behaviour above again.

**Comment by pearcetm** [comment]
This is strange, it is like your setup does the opposite of what the more basic viewer does. #2537 addressed an issue where the tiling-during-loading was happening even though `immediateRender === false`, and now you're once again seeing behavior this when others aren't.
